src/test/java/cloud/fogbow/ras/core/plugins/interoperability/aws/attachment/v2/AwsV2AttachmentPluginTest.java

```diff
@@ -119,159 +108,275 @@ public void testHasFailed() {
+        AwsV2User cloudUser = Mockito.mock(AwsV2User.class);
 
-		// exercise
-		AttachmentInstance attachmentInstance = this.plugin.getInstance(attachmentOrder, cloudUser);
+        // exercise
+        this.plugin.getInstance(order, cloudUser);
 
-		// verify
-		PowerMockito.verifyStatic(AwsV2ClientUtil.class, VerificationModeFactory.times(1));
-		AwsV2ClientUtil.createEc2Client(Mockito.anyString(), Mockito.anyString());
+        // verify
+        PowerMockito.verifyStatic(AwsV2ClientUtil.class, VerificationModeFactory.times(TestUtils.RUN_ONCE));
+        AwsV2ClientUtil.createEc2Client(Mockito.eq(cloudUser.getToken()), Mockito.anyString());
 
-		Mockito.verify(client, Mockito.times(1)).describeVolumes(Mockito.any(DescribeVolumesRequest.class));
+        PowerMockito.verifyStatic(AwsV2CloudUtil.class, VerificationModeFactory.times(TestUtils.RUN_ONCE));
+        AwsV2CloudUtil.doDescribeVolumesRequest(Mockito.eq(request), Mockito.eq(client));
 
-		Mockito.verify(this.plugin, Mockito.times(1))
-				.buildAttachmentInstance(Mockito.any(DescribeVolumesResponse.class));
+        Mockito.verify(this.plugin, Mockito.times(TestUtils.RUN_ONCE))
+                .buildAttachmentInstance(Mockito.any(DescribeVolumesResponse.class));
+    }
+    
+    // test case: When calling the doDeleteInstance method, it must verify
+    // that is call was successful.
+    @Test
+    public void testDoDeleteInstance() throws FogbowException {
+        // set up
+        Ec2Client client = this.testUtils.getAwsMockedClient();
+        String volumeId = TestUtils.FAKE_VOLUME_ID;
 
-		Assert.assertEquals(expected, attachmentInstance);
-	}
+        DetachVolumeRequest request = DetachVolumeRequest.builder()
+                .volumeId(volumeId)
+                .build();
 
-	// test case: When calling the defineDeviceNameAttached method, with a device
-	// name null, it must return a default device name.
-	@Test
-	public void testDefineDeviceNameAttachedWithNullValue() {
-		// set up
-		String expected = AwsV2AttachmentPlugin.DEFAULT_DEVICE_NAME;
+        DetachVolumeResponse response = DetachVolumeResponse.builder().build();
+        Mockito.when(client.detachVolume(Mockito.eq(request))).thenReturn(response);
 
-		// exercise
-		String actual = this.plugin.getAttachedDeviceName(null);
+        // exercise
+        this.plugin.doDeleteInstance(volumeId, request, client);
 
-		// verify
-		Assert.assertEquals(expected, actual);
-	}
-	
-	// test case: When calling the defineDeviceNameAttached method, with an empty
-	// device name, it must return a default device name.
-	@Test
-	public void testDefineDeviceNameAttachedWithEmptyString() {
-		// set up
-		String expected = AwsV2AttachmentPlugin.DEFAULT_DEVICE_NAME;
+        // verify
+        Mockito.verify(client, Mockito.times(TestUtils.RUN_ONCE)).detachVolume(Mockito.eq(request));
+    }
+    
+    // test case: When calling the defineDeviceNameAttached method, and an
```

Comment:
This description doesn't match the method being tested.

Reply:
Same for testGetDeviceNameAttachedWithNullValue()

Reply:
Okay, I'll fix it.